Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift

```diff
@@ -0,0 +1,188 @@
+// Copyright (C) 2024 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+
+#if canImport(LinearMediaKit)
+
+import LinearMediaKit
+import WebKitSwift
+
+// MARK: Objective-C Implementations
+
+@_objcImplementation extension WKSLinearMediaTimeRange {
+    var lowerBound: TimeInterval
+    let upperBound: TimeInterval
+
+    init(lowerBound: TimeInterval, upperBound: TimeInterval) {
+        self.lowerBound = lowerBound
+        self.upperBound = upperBound
+    }
+}
+
+@_objcImplementation extension WKSLinearMediaTrack {
+    let localizedDisplayName: String
+
+    init(localizedDisplayName: String) {
+        self.localizedDisplayName = localizedDisplayName
+    }
+}
+
+// MARK: LinearMediaKit Extensions
+
+extension WKSLinearMediaContentMode {
+    init(_ contentMode: ContentMode?) {
+        guard let contentMode else {
+            self = .none
+            return
+        }
+
+        switch contentMode {
+        case .scaleAspectFit:
+            self = .scaleAspectFit
+        case .scaleAspectFill:
+            self = .scaleAspectFill
+        case .scaleToFill:
+            self = .scaleToFill
+        @unknown default:
+            fatalError()
+        }
```

Comment:
```suggestion
        self = switch contentMode {
        case .scaleAspectFit?:
            .scaleAspectFit
        case .scaleAspectFill?:
           .scaleAspectFill
        case .scaleToFill?:
            .scaleToFill
         case .none:
            .none
        @unknown default:
            fatalError()
        }
```
maybe?

Reply:
whoa, cool

Reply:
Darn, unfortunately this syntax isn't supported by the swift that ships on Monterey:
```
/Volumes/Data/worker/macOS-Monterey-Release-Build-EWS/build/Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift:128:16: error: expected expression in assignment
        self = switch viewingMode {
```

Reply:
Actually, why is this code even building on macOS Monterey? Why would `canImport(LinearMediaKit)` be true there? 🤔